src/main/java/edu/escuelaing/ieti/matchmaking/services/impl/UserServiceImpl.java

```diff
@@ -9,35 +9,49 @@
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Optional;
 
 @Service
 public class UserServiceImpl implements UserService {
 
     @Autowired
-    private UserRepository inMemoryRepository;
+    private UserRepository userRepository;
 
     @Override
     public User create(User user) throws EntityExistsException {
-        return inMemoryRepository.create(user);
+        if (userRepository.existsByEmail(user.getEmail())){
+            throw new EntityExistsException(User.class, "User email", user.getEmail());
+        }
+        return userRepository.save(user);
     }
 
     @Override
-    public void remove(String userId) throws EntityNotFoundException {
-        inMemoryRepository.remove(userId);
+    public void remove(User user) throws EntityNotFoundException {
+        if (!userRepository.existsByEmail(user.getEmail())){
+            throw new EntityNotFoundException(User.class, "User email", user.getEmail());
+        }
+        userRepository.delete(user);
     }
 
     @Override
     public User update(User user) throws EntityNotFoundException {
-        return inMemoryRepository.update(user);
+        if (!userRepository.existsByEmail(user.getEmail())){
+            throw new EntityNotFoundException(User.class, "User email", user.getEmail());
+        }
+        return userRepository.save(user);
     }
 
     @Override
     public User getUserById(String userId) throws EntityNotFoundException {
-        return inMemoryRepository.getById(userId);
+        Optional<User> optionalUser = userRepository.findById(userId);
+        if (!optionalUser.isPresent()){
```

Comment:
This is not the proper use of the `Optional` type. Take a look into the patterns for `Optional` usage in the **Effective Java 3rd Edition** book.